content/en/docs/concepts/cluster-administration/node-autoscaling.md

```diff
@@ -0,0 +1,264 @@
+minimum limit).
+
+The specifics of the provisioned Nodes (for example the amount of resources, the presence of a given label) depend on
+autoscaler configuration. Autoscalers can either choose them from a pre-defined set of Node configurations,
+or use [auto-provisioning](#autoprovisioning).
+
+{{< note >}}
+Provisioning is formerly known as _scale-up_ in Cluster Autoscaler.
+{{< /note >}}
+
+### Auto-provisioning {#autoprovisioning}
+
+Node auto-provisioning is a mode of provisioning in which a user doesn't have
+to fully configure the specifics of the Nodes that can be provisioned. Instead,
+the autoscaler dynamically chooses the Node configuration based on the pending Pods
+it's reacting to, as well as pre-configured constraints (for example the minimum amount
+of resources, the need for a given label).
+
+## Node consolidation {#consolidation}
+
+The main component of the cost of running a cluster is usually the cost of the
+VMs backing its Nodes. In order to minimize this cost and still be able to
+run all Pods in the cluster, the Pods should utilize as much of the Nodes'
+resources as possible. From this perspective, the overall Node utilization in
+a cluster can be used as a proxy for how cost-effective the cluster is.
+
+{{< note >}}
+For the purpose of consolidation, resource utilization is computed using Pod resource requests
+and Node allocatable resources. Real resource usage is not taken into account. The details of
+how utilization is computed can vary between autoscalers.
+{{< /note >}}
+
+Nodes in your cluster can be automatically _consolidated_ in order
+to improve the overall Node utilization, and in turn the cost-effectiveness of
+the cluster.
+
+Consolidation happens through removing a set of suboptimally utilized Nodes from
+the cluster. Optionally, a different set of Nodes can be [provisioned](#provisioning) to replace
+them.
+
+For the purpose of consolidation, a Node is considered _empty_ if it only has
+DaemonSet and static Pods running on it. Removing empty Nodes during consolidation
+is more straightforward than non-empty ones, and autoscalers often have optimizations
+designed specifically for consolidating empty nodes.
+
+Removing non-empty Nodes during consolidation is disruptive - the Pods running on them
+are terminated, and possibly have to be recreated (for example by a Deployment). However, all such
+recreated pods should be able to schedule on existing Nodes in the cluster, or the
+replacement Nodes provisioned as part of consolidation. No pods should
+become pending as a result of consolidation.
+
+{{< note >}}
+Consolidation is prone to certain race conditions. For example, if a new Pod is
+created right after a Node is consolidated, it can get scheduled on an existing Node
+in the cluster instead of the Pods recreated because of consolidation - leaving
+these Pods pending. This scenario is usually automatically recovered by provisioning.
+{{< /note >}}
```

Comment:
Ah I got it now. 

WDYT about this? 

In consolidating nodes, autoscalers try to simulate how the kube-scheduler will assign pods that are rescheduling. Since autoscalers do not control the actual scheduling process, autoscalers cannot guarantee that "no pods should become pending as a result of consolidation". This can change based on the churn in the cluster, the complexity of scheduling constraints, and other autoscaler-specific constraints.